### ISSUE TITLE
Warning on connections in MultiBody model from the frontend when translated with the new backend

### ISSUE THREAD

**Issue by anotheruserofgithub** [OpenModelica/OpenModelica#12590]
Please consider the following N-pendulum model (adapted from https://github.com/OpenModelica/OpenModelica/pull/10471#issuecomment-1753193826):
```modelica
model NPendulum
  constant Integer N = 1;
  Modelica.Mechanics.MultiBody.Joints.Revolute[N] rev(each phi(fixed = true), each w(fixed = true));
  Modelica.Mechanics.MultiBody.Parts.BodyBox[N] box(each r = {1, 0, 0});
  inner Modelica.Mechanics.MultiBody.World world(animateWorld = false, animateGravity = false);
equation
  assert(N > 0, "N must be greater than zero");
  connect(world.frame_b, rev[1].frame_a);
  for i in 1:N loop
    connect(rev[i].frame_b, box[i].frame_a);
  end for;
  for i in 1:N-1 loop
    connect(box[i].frame_b, rev[i+1].frame_a);
  end for;
end NPendulum;
```

When you translate with the new backend (`--newBackend`), although it is not yet able to handle this model even in the simplest case `N = 1`, you run into a warning in the frontend:
```
[Modelica.Mechanics.MultiBody.Joints.Revolute: 102:3-112:9]: [Connections.rooted(rev.frame_a.R) needs exactly one statement Connections.branch(rev.frame_a.R, B.R) involving rev.frame_a.R but we found none in the graph. Run with -d=cgraphGraphVizFile to debug](omeditmessagesbrowser:///Modelica.Mechanics.MultiBody.Joints.Revolute?lineNumber=102)
```

With the debug flag `-d=cgraphGraphVizFile` and then the command `dot -Tsvg NPendulum.gv > NPendulum.svg` I was able to generate this graph:
![NPendulum](https://github.com/OpenModelica/OpenModelica/assets/96748782/1aa62f88-1df8-475f-a66b-9bad3e5cfaa4)

Is anything wrong in the model? The warning occurs whatever the value of `N`. There is no such message with the old backend.

**Comment by casella** [comment]
@anotheruserofgithub it is really too early to try the new backend on MultiBody systems, because we still lack proper index reduction support.

I'm not sure if the new backend already handles overconstrained connectors, if so, this issue should not be there. @kabdelhak, @phannebohm can you comment on that?

**Comment by casella** [comment]
Or maybe it's a front-end issue, @perost?

**Comment by perost** [comment]
> Or maybe it's a front-end issue, @perost?

Yeah, it's almost certainly a frontend issue. The frontend behaves differently with the new backend due to all the array-preserving stuff, and a lot of things don't work with it.

**Comment by casella** [comment]
OK. MultiBody is probably still a long way to go with the NB, but PowerGrids 2.0 uses overconstrained connectors and should be supported ASAP. Worth having a look, can be fixed in August if it's not a trivial fix. Thanks!

**Comment by casella** [comment]
> Assuming that [PowerGrids_dev](https://libraries.openmodelica.org/branches/newInst-newBackend/PowerGrids_dev/PowerGrids_dev.html) in the library tests is PowerGrids 2.0

Yes, the assumption is correct 😃 

> Then all the models pass the frontend with the new backend.

OK, good.
